### ISSUE TITLE
[api-extractor] Type-only exports missing in .d.ts rollup

### ISSUE THREAD

**Issue by MattiasBuelens** [microsoft/rushstack#1760]
**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**

[Type-only exports](https://devblogs.microsoft.com/typescript/announcing-typescript-3-8/#type-only-imports-exports) are missing in the generated .d.ts rollup.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

* src/index.ts
```typescript
import {Bar, Foo} from "./foo";

export {Foo};

export type {Bar};
```
* src/foo.ts
```typescript
/** @public */
export class Foo {
    greet() {
        console.log('Hello from foo!');
    }

    createBar(): Bar {
        return new Bar();
    }
}

/** @public */
export class Bar {
    greet() {
        console.log('Hello from bar!');
    }
}
```

Generated .d.ts rollup:
```typescript
/** @public */
declare class Bar {
    greet(): void;
}

/** @public */
export declare class Foo {
    greet(): void;
    createBar(): Bar;
}

export { }
```

Repository with minimal reproduction: [api-extractor-type-only-export](https://github.com/MattiasBuelens/api-extractor-type-only-export/tree/7567aaaa129651f48b202e42fed1f9810edf33c7)

**What is the expected behavior?**

The type-only exports are retained.
```typescript
/** @public */
declare class Bar {
    greet(): void;
}

/** @public */
export declare class Foo {
    greet(): void;
    createBar(): Bar;
}

export type { Bar }; // <<< should be retained!

export { }
```

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** api-extractor
* **Tool Version:** 7.7.8
* **Node Version:** 10.15.3
  * **Is this a LTS version?** Yes
  * **Have you tested on a LTS version?** Yes
* **OS:** Windows 10 (1909)

---

**Notes:**

I'm guessing that since [API Extractor 7.7.8 is still on TypeScript 3.7.2](https://github.com/microsoft/rushstack/blob/%40microsoft/api-extractor_v7.7.8/apps/api-extractor/package.json#L43), it doesn't yet understand type-only imports and exports which were introduced in TypeScript 3.8. Are there any plans to update to the latest version of TypeScript, and update the rollup logic to support these new imports and exports?

**Comment by csr632** [comment]
Also discuession here: https://github.com/microsoft/rushstack/issues/1866

**Comment by octogonz** [comment]
This is a recent TypeScript language feature that we need to support.

**Comment by octogonz** [comment]
This should be fixed by #1985

**Comment by octogonz** [comment]
This fix was released with API Extractor 3.9.0
